utilities/test_suite/rpp_test_suite_common.h

```diff
@@ -968,11 +968,47 @@ void compare_outputs_pln(Rpp8u* output, Rpp8u* refOutput, RpptDescPtr dstDescPtr
                     outRefVal = rowTempRef + j ;
                     int diff = abs(*outVal - *outRefVal);
                     if(diff <= CUTOFF)
-                        matched_idx++;
+                        matchedIdx++;
                 }
             }
         }
-        if(matched_idx == (height * width * dstDescPtr->c) && matched_idx !=0)
+        if(matchedIdx == (height * width * dstDescPtr->c) && matchedIdx !=0)
+            fileMatch++;
+    }
+}
+
+// compares the output of PLN3-PLN3 variants
+void compare_outputs_pln3(Rpp8u* output, Rpp8u* refOutput, RpptDescPtr dstDescPtr, RpptImagePatch *dstImgSizes, int refOutputHeight, int refOutputWidth, int refOutputSize, int &fileMatch)
+{
+    Rpp8u *rowTemp, *rowTempRef, *outVal, *outRefVal, *outputTemp, *outputTempRef, *outputTempChn, *outputTempRefChn;
+    for(int imageCnt = 0; imageCnt < dstDescPtr->n; imageCnt++)
+    {
+        outputTemp = output + imageCnt * dstDescPtr->strides.nStride;
+        outputTempRef = refOutput + imageCnt * refOutputSize;
+        int height = dstImgSizes[imageCnt].height;
+        int width = dstImgSizes[imageCnt].width;
+        int matchedIdx = 0;
+        int refOutputHstride = refOutputWidth * dstDescPtr->c;
+
+        for(int c = 0; c < dstDescPtr->c; c++)
+        {
+            outputTempChn = outputTemp + c * dstDescPtr->strides.cStride;
+            outputTempRefChn = outputTempRef + c;
+            for(int i = 0; i < height; i++)
+            {
+                rowTemp = outputTempChn + i * dstDescPtr->strides.hStride;
+                rowTempRef = outputTempRefChn + i * refOutputHstride;
+                for(int j = 0; j < width; j++)
+                {
+                    outVal = rowTemp + j;
+                    outRefVal = rowTempRef + j * 3 ;
```

Comment:
Remove space before semicolon at L1004
Also better to add a comment that reference output is PLN3 and test suite output is in PKD3 for better clarity?

@r-abishek Please share your thoughts also on this

Reply:
Done